### PR TITLE
🐛 adds tests for S3TransferError; refactoring flaky CI test; better logging for long running task errors

### PR DESCRIPTION
**Pull request by GitHK**
## What do these changes do?

- Adds tests for `_check_for_aws_http_errors`.
- adjusts timeout detection interval in `tests/fastapi/long_running_tasks/test_long_running_tasks_context_manager.py::test_task_result_times_out` (**lesson learned:**  when dealing with timeouts always use an order of magnitude of difference in checks)
- enhances logging when long running fastapi tasks finished with error

## Related issue/s

- addresses `tests/fastapi/long_running_tasks/test_long_running_tasks_context_manager.py::test_task_result_times_out` in  https://github.com/ITISFoundation/osparc-simcore/issues/3421
- will help figuring out S3TransferErrors https://github.com/ITISFoundation/osparc-simcore/issues/3531

## Checklist

- [x] Unit tests for the changes exist

### CODE REVIEW

File: packages/simcore-sdk/tests/unit/test_node_ports_common_file_io_utils.py [hunk 1]
```diff
@@ -6,26 +6,100 @@
...
 
-TEST_ERROR = "OPSIE there was an error here"
-
 
-@pytest.fixture
-def test_client(loop: BaseEventLoop, aiohttp_client) -> TestClient:
-    async def raise_error(request):
-        return web.Response(body=TEST_ERROR, status=web.HTTPBadRequest.status_code)
+def _get_test_client(
+    mocked_response: web.Response,
```
Comment (pcrespov) [inline packages/simcore-sdk/tests/unit/test_node_ports_common_file_io_utils.py:18]: As agreed, try with ``aioresponse`` fixtures

**Review by pcrespov — COMMENTED** [review]
Can you also please add a more descriptive title for the PR

File: packages/simcore-sdk/tests/unit/test_node_ports_common_file_io_utils.py [hunk 2]
```diff
@@ -1,31 +1,82 @@
...
+def mock_service() -> Iterable[aioresponses]:
+    with aioresponses() as mock:
+        yield mock
 
-    app = web.Application()
-    app.router.add_get("/", raise_error)
-    return loop.run_until_complete(aiohttp_client(app))
 
+@pytest.fixture
+async def aiohttp_client() -> AsyncIterable[ClientSession]:
```
Comment (sanderegg) [inline packages/simcore-sdk/tests/unit/test_node_ports_common_file_io_utils.py:25]: Question, is that not already a fixture from pytest-aiohttp????
why not directly use it?

File: packages/simcore-sdk/tests/unit/test_node_ports_common_file_io_utils.py [hunk 3]
```diff
@@ -1,31 +1,82 @@
...
+            "<RequestId>7EE901348D6C6812</RequestId><HostId>"
+            "FfQE7jdbUt39E6mcQq/"
+            "ZeNR52ghjv60fccNT4gCE4IranXjsGLG+L6FUyiIxx1tAuXL9xtz2NAY7ZlbzMTm94fhY3TBiCBmf"
+            "</HostId></Error>",
+        ),
+        (500, ""),
+        (503, ""),
+    ],
+)
+async def testtest_check_for_aws_http_errors_true_raise_for_status(
```
Comment (sanderegg) [inline packages/simcore-sdk/tests/unit/test_node_ports_common_file_io_utils.py:60]:
```suggestion
async def test_check_aws_http_errors_raises(
```

**Review by sanderegg — COMMENTED** [review]
pair reviewd

File: packages/simcore-sdk/tests/unit/test_node_ports_common_file_io_utils.py [hunk 4]
```diff
@@ -1,31 +1,82 @@
...
-TEST_ERROR = "OPSIE there was an error here"
+A_TEST_ROUTE = "http://127.0.0.1:1249/test-route"
 
 
 @pytest.fixture
-def test_client(loop: BaseEventLoop, aiohttp_client) -> TestClient:
-    async def raise_error(request):
-        return web.Response(body=TEST_ERROR, status=web.HTTPBadRequest.status_code)
+def mock_service() -> Iterable[aioresponses]:
+    with aioresponses() as mock:
```
Comment (pcrespov) [inline packages/simcore-sdk/tests/unit/test_node_ports_common_file_io_utils.py:20]: We already have a fixture that does this in ``pytest_simcore``:

https://github.com/ITISFoundation/osparc-simcore/blob/3f88ad3d6340b40d616db8c4d0eee80d47da44ed/packages/pytest-simcore/src/pytest_simcore/aioresponses_mocker.py#L17

Comment (GitHK) [reply packages/simcore-sdk/tests/unit/test_node_ports_common_file_io_utils.py:21]: I seem to have missed it, even after looking at it :\

File: packages/simcore-sdk/tests/unit/test_node_ports_common_file_io_utils.py (hunk 2 above)
Comment (GitHK) [reply packages/simcore-sdk/tests/unit/test_node_ports_common_file_io_utils.py]: Made a mess with the names. I was looking for something else. I required an `aiohttp.ClientSession`

**Review by pcrespov — APPROVED** [review]

**Review by pcrespov — APPROVED** [review]

**Review by sanderegg — APPROVED** [review]

**Pull request merged**